src/clock/mumemoment.cpp

```diff
@@ -146,133 +130,175 @@ MumeTimeEnum MumeMoment::toTimeOfDay() const
     return MumeTimeEnum::DAY;
 }
 
-int MumeMoment::moonCycle() const
+int MumeMoment::moonZenithMinutes() const
 {
-    // Moon cycle is 24 days long and starts with the ending of the new moon
-    return (year * MUME_DAYS_PER_YEAR + month * MUME_DAYS_PER_MONTH + day)
-           % MUME_DAYS_PER_MOON_CYCLE;
+    // At what minute of day the moon is at its highest point in the sky
+    const auto count = static_cast<int64_t>(toSeconds()) * MUME_MINUTES_PER_DAY;
+    return (count / MUME_MINUTES_PER_MOON_CYCLE) % MUME_MINUTES_PER_DAY;
 }
 
-int MumeMoment::moonRise() const
+MumeMoonPositionEnum MumeMoment::moonPosition() const
 {
-    return (moonCycle() + 6) % MUME_DAYS_PER_MOON_CYCLE;
+    int rise = moonZenithMinutes() + MUME_MINUTES_PER_DAY * 3 / 4;
+    if (rise >= MUME_MINUTES_PER_DAY)
+        rise -= MUME_MINUTES_PER_DAY;
+    int now = minute + hour * MUME_MINUTES_PER_HOUR;
+    if (now < rise)
+        now += MUME_MINUTES_PER_DAY;
+    /* dir 0-15: 0: east, 1/2: southeast, 3/4: south, 5/6: southwest; 7: west;
+           8-15: under the horizon*/
+    const int dir = std::clamp((now - rise) * 16 / MUME_MINUTES_PER_DAY, 0, 15);
+    if (dir < 8) {
+        static const constexpr int EAST = static_cast<int>(MumeMoonPositionEnum::EAST);
+        static const constexpr int MIN = static_cast<int>(MumeMoonPositionEnum::INVISIBLE);
+        static const constexpr int MAX = static_cast<int>(MumeMoonPositionEnum::WEST);
+        return static_cast<MumeMoonPositionEnum>(std::clamp(EAST + (dir + 1) / 2, MIN, MAX));
+    }
+    return MumeMoonPositionEnum::INVISIBLE;
 }
 
-int MumeMoment::moonSet() const
+int MumeMoment::moonLevel() const
 {
-    return (moonCycle() + 18) % MUME_DAYS_PER_MOON_CYCLE;
+    // Illumination level seems to be between 0-12
+    // 0 is new moon, 12 is full moon
+    // Levels > 4 make the moon show up in the prompt as ')'
+    int level = (moonZenithMinutes() + MUME_MINUTES_PER_HOUR / 2) / MUME_MINUTES_PER_HOUR;
```

Comment:
Hmmm, this is a little cryptic to me - why add MUME_MINUTES_PER_HOUR /2 in the first "term" here? What does that do?

Reply:
To be honest, this is just what I could get working. Cryptic is the name of reverse engineering and in this case I tried re-using some other mechanisms but it wouldn't be as stable in synchronizing.